pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -262,11 +279,13 @@ func (c *FirstContainerReady) Accept(deployment *kapi.ReplicationController) err
 		return fmt.Errorf("no pods found for deployment %s", deployutil.LabelForDeployment(deployment))
 	}
 
-	// If we found multiple, use the first one and log a warning.
-	// TODO: should finding multiple be an error?
-	pod := &pods.Items[0]
-	if len(pods.Items) > 1 {
-		glog.Infof("Warning: more than one pod for deployment %s; basing canary check on the first pod '%s'", deployutil.LabelForDeployment(deployment), pod.Name)
+	// Use the first unknown pod we find, and track all pods we see.
+	var pod *kapi.Pod
+	for _, candidate := range pods.Items {
+		if !c.knownPods.Has(candidate.Name) && pod == nil {
+			pod = &candidate
```

Comment:
Why don't we break here?  We're going to check only one pod, but claim we know about all of the pods?